lms/djangoapps/course_structure_api/v0/views.py

```diff
@@ -0,0 +1,236 @@
+
+    def check_course_permissions(self, request, course):
+        """
+        Checks if the request user can access the course.
+        Raises PermissionDenied if the user does not have course acccess.
+        """
+        if not self.user_can_access_course(request.user, course):
+            raise PermissionDenied
+
+
+class CourseViewMixin(AuthMixin):
+    """
+    Mixin for views dealing with course content.
+    """
+    lookup_field = 'course_id'
+
+    def get_serializer_context(self):
+        """
+        Supplies a course_id to the serializer.
+        """
+        context = super(CourseViewMixin, self).get_serializer_context()
+        context['course_id'] = self.kwargs.get('course_id')
+        return context
+
+    def get_course_or_404(self):
+        """
+        Retrieves the specified course, or raises an Http404 error if it does not exist.
+        Also checks to ensure the user has permissions to view the course
+        """
+        try:
+            course_id = self.kwargs.get('course_id')
+            course_key = CourseKey.from_string(course_id)
+            course = courses.get_course(course_key)
+
+            self.check_course_permissions(self.request, course)
+
+            return course
+        except ValueError:
+            raise Http404
+
+
+class CourseList(CourseViewMixin, ListAPIView):
+    """
+    **Use Case**
+
+        CourseList returns paginated list of courses in the edX Platform. The list can be
+        filtered by course_id
+
+    **Example Request**
+
+          GET /
+          GET /?course_id={course_id1},{course_id2}
+
+    **Response Values**
+
+        * category: The type of content. In this case, the value is always "course".
+
+        * name: The name of the course.
+
+        * uri: The URI to use to get details of the course.
+
+        * course: The course number.
+
+        * due:  The due date. For courses, the value is always null.
+
+        * org: The organization specified for the course.
+
+        * id: The unique identifier for the course.
+    """
+    paginate_by = 10
+    paginate_by_param = 'page_size'
+    pagination_serializer_class = PaginationSerializer
+    serializer_class = serializers.CourseSerializer
+
+    def get_queryset(self):
+        course_ids = self.request.QUERY_PARAMS.get('course_id', None)
+
+        course_descriptors = []
+        if course_ids:
+            course_ids = course_ids.split(',')
+            for course_id in course_ids:
+                course_key = CourseKey.from_string(course_id)
+                course_descriptor = courses.get_course(course_key)
+                course_descriptors.append(course_descriptor)
+        else:
+            course_descriptors = modulestore().get_courses()
+
+        results = [course for course in course_descriptors if self.user_can_access_course(self.request.user, course)]
+
+        if not settings.DEBUG and len(results) == 0:
+            raise PermissionDenied
```

Comment:
I was making an assumption around reasons for an empty list: user cannot access any courses. This is a flawed assumption. Returning empty list.